core/src/main/scala/com/rallyhealth/vapors/factfilter/data/FactType.scala

```diff
@@ -1,16 +1,25 @@
 package com.rallyhealth.vapors.factfilter.data
 
 import cats.Order
-import com.rallyhealth.vapors.core.data.NamedLens
+import com.rallyhealth.vapors.core.data.{NamedLens, ValidDataPathKey}
 
 import scala.reflect.ClassTag
 import scala.reflect.runtime.universe.TypeTag
 
+/**
+  * Type information required to construct a [[Fact]].
+  */
 trait FactType[T] {
 
+  /**
+    * The unique name for this fact type.
+    */
   def name: String
 
-  final lazy val fullName: String = s"$name: ${tt.tpe}"
+  /**
+    * The unique type name plus type information.
+    */
+  final lazy val fullName: String = s"'$name' as ${tt.tpe}"
```

Comment:
Could we give this a type besides String?

Reply:
Because it's used in FactTable and probably other places.

Reply:
I think we only use FactType in the public API. The use of the name as a `String` should only be used for implementation. I'll double check on this.